src/measure_theory/decomposition/signed_hahn.lean

```diff
@@ -0,0 +1,597 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.vector_measure
+import order.symm_diff
+
+/-!
+# Hahn decomposition
+
+This file defines positive and negative sets with respect to a vector measure and
+prove the Hahn decomposition (signed version).
+
+## Main definitions
+
+* `vector_measure.positive` : a set `i` is positive with respect to a vector measure if for all
+  measurable sets `j` contained in `i`, the measure of `j` is non-negative.
+* `vector_measure.negative` : a set `i` is negative with respect to a vector measure if for all
+  measurable sets `j` contained in `i`, the measure of `j` is non-positive.
+
+## Main results
+
+* `signed_measure.exists_disjoint_positive_negative_union_eq` : the Hahn decomposition theorem.
+* `signed_measure.exists_negative_set` : A measurable set of negative measure contains at least
+  one negative subset.
+
+## Tags
+
+positive, negative, Hahn decomposition
+-/
+
+noncomputable theory
+open_locale classical big_operators nnreal ennreal
+
+section lemmas
+
+variables {α : Type*}
+
+open set filter
+
+lemma exists_tendsto_Inf {S : set ℝ} (hS : ∃ x, x ∈ S) (hS' : ∃ x, ∀ y ∈ S, x ≤ y) :
+  ∃ (f : ℕ → ℝ) (hf : ∀ n, f n ∈ S), tendsto f at_top (nhds (Inf S)) :=
+begin
+  have : ∀ n : ℕ, ∃ t ∈ S, t < Inf S + 1 / (n + 1 : ℝ),
+  { exact λ n, (real.Inf_lt _ hS hS').1 ((lt_add_iff_pos_right _).2 nat.one_div_pos_of_nat) },
+  choose f hf using this,
+  refine ⟨f, λ n, (hf n).1, _⟩,
+  rw tendsto_iff_dist_tendsto_zero,
+  refine squeeze_zero_norm _ tendsto_one_div_add_at_top_nhds_0_nat,
+  intro n,
+  obtain ⟨hf₁, hf₂⟩ := hf n,
+  rw [real.dist_eq, real.norm_eq_abs, abs_abs,
+      abs_of_nonneg (sub_nonneg.2 (real.Inf_le S hS' hf₁))],
+  linarith,
+end
+
+lemma exists_tendsto_Sup {S : set ℝ} (hS : ∃ x, x ∈ S) (hS' : ∃ x, ∀ y ∈ S, y ≤ x) :
```

Comment:
I'm not sure where this and the next lemma should go. I'm thinking either `analysis.specific_limits` or `topology.instances.real`.

Reply:
To me these lemmas are close friends of `exists_seq_strict_mono_tendsto'` (maybe it's even possible to refactor things to get a common generalization of the two of them). So I'd put them in `topology/algebra/ordered/basic`.

Reply:
I can't seem to find `exists_seq_strict_mono_tendsto'` on mathlib docs, is this a new lemma?

Reply:
Sorry, my bad, it's only available in #8266. I can try to extract a common generalization of your lemma and the lemma in there in another PR, if you want.

Reply:
Would you there is an easy way of using `exists_seq_strict_mono_tendsto'` here? I tried a bit but I don't think it's possible since a subset of an order topology doesn't have to be an order topology